dts/bindings/i2s/st,stm32-i2s.yaml

```diff
@@ -25,3 +25,12 @@ properties:
 
   pinctrl-names:
     required: true
+
+  mck-enabled:
+    type: phandle
+    description: |
+      Designated pin to enable the Master Clock Output alternate function.
+      Successful enabling requires the phandle to also be listed in the
+      pinctrl-0 array.
+      The onus is on the user to ensure that the phandle is accurate and
+      thereby points to the actual pin function matching the hardware.
```

Comment:
why isn't this a simple boolean?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/54880#discussion_r1136029326

Reply:
I still don't see the value of doing this. If we want to do some sort of compile time checks, there are better ways to do that (extend edtlib).

Reply:
Bear with me on this one, I am not familiar with edtlib. Could you please provide some insights, perhaps most beneficial if geared towards the specifics of the problem at hand, on how to validate that any of the multiple i2sX_mck_pins are part of `pinctrl-0` when mck-enabled is present? 

I am conscious of what has been mentioned in regards to the hardware specificity handling in DT, is it fair to assume that edtlib can cover a case that is narrowed down to the STM32 series?

Reply:
there's nothing in place now to do that, but I guess edtlib (see in scripts/dts/python-devicetree/) could be used to have some post-configure stage checks. That is, when you get the final zephyr.dts, have vendor specific hooks that run and do random checks, or even create utilities like a script to plot assigned pins, etc.

I'd consider this a secondary task to this PR. It'd probably involve discussing it at the Arch WG as well.

But the general idea I'd like to transmit here is that abusing DT properties/using excessive macrology for the sake of having static analysis, is, in my opinion, not the right solution.

Reply:
Thanks for the explanation, it makes good sense to me. It would be rather nice if such a configuration check existed, perhaps even generalised sufficiently to ensure that the compiled DT contained a minimal pinctrl combination of functions for the different enabled peripherals.

Let me know if it is worth raising an issue/suggestion, or some other means, to provide suitable visibility for future consideration.

For now, I'll simplify the property to a boolean type to cover the immediate need.